test/integration/src/test/java/io/pravega/test/integration/LargeEventTest.java

```diff
@@ -0,0 +1,510 @@
+             ReaderGroupManager readerGroupManager = new ReaderGroupManagerImpl(SCOPE_NAME, controller, clientFactory)) {
+            // Start writing events to the stream.
+            val writers = createEventWriters(streamName, NUM_WRITERS, clientFactory);
+            // Create a ReaderGroup.
+            createReaderGroup(readerGroupName, readerGroupManager, streamName);
+            // Create Readers.
+            val readers = createEventReaders(NUM_READERS, clientFactory, readerGroupName);
+            Futures.allOf(writers).get();
+            stopReadFlag.set(true);
+
+            this.server.close();
+            // Reset the server, in effect clearing the AppendProcessor and PravegaRequestProcessor.
+            this.server = new PravegaConnectionListener(false, servicePort, store, tableStore, serviceBuilder.getLowPriorityExecutor());
+            this.server.startListening();
+            Futures.allOf(readers).get();
+
+            log.info("Deleting ReaderGroup: {}", readerGroupName);
+            readerGroupManager.deleteReaderGroup(readerGroupName);
+        }
+
+        validateCleanUp(streamName);
+        validateEventReads(eventsReadFromPravega, eventsWrittenToPravega);
+    }
+
+    @Test
+    public void testReadWriteWithConnectionReconnect() throws ExecutionException, InterruptedException {
+        String readerGroupName = "testLargeEventReconnectReaderGroup";
+        String streamName = "ConnectionReconnect";
+        StreamConfiguration config = getStreamConfiguraton(NUM_READERS);
+
+        createScopeStream(SCOPE_NAME, streamName, config);
+        generateWriteEventData(NUM_WRITERS, LARGE_EVENT_SIZE);
+
+        try (ConnectionExporter connectionFactory = new ConnectionExporter(ClientConfig.builder().build());
+             ClientFactoryImpl clientFactory = new ClientFactoryImpl(SCOPE_NAME, controller, connectionFactory);
+             ReaderGroupManager readerGroupManager = new ReaderGroupManagerImpl(SCOPE_NAME, controller, clientFactory)) {
+            // Start writing events to the stream.
+            val writers = createReconnectingEventWriters(streamName, NUM_WRITERS, clientFactory, connectionFactory);
+            // Create a ReaderGroup.
+            createReaderGroup(readerGroupName, readerGroupManager, streamName);
+            // Create Readers.
+            val readers = createEventReaders(NUM_READERS, clientFactory, readerGroupName);
+            Futures.allOf(writers).get();
+            stopReadFlag.set(true);
+
+            Futures.allOf(readers).get();
+
+            log.info("Deleting ReaderGroup: {}", readerGroupName);
+            readerGroupManager.deleteReaderGroup(readerGroupName);
+        }
+
+        validateCleanUp(streamName);
+        validateEventReads(eventsReadFromPravega, eventsWrittenToPravega);
+    }
+
+    @Test
+    public void testReadWriteStreamSeal() throws ExecutionException, InterruptedException {
```

Comment:
I don't see the stream being sealed anywhere. Shouldn't this confirm the correct exception is thrown when writing to a sealed stream?